server/api/user/user.model.ts

```diff
@@ -49,41 +49,74 @@ export class User extends BaseEntity {
   @OneToMany(() => UserRole, userRole => userRole.user)
   userRoles!: UserRole[];
 
-  async addRole(entityManager: EntityManager, role: Role, indexPrefix?: string): Promise<UserRole> {
-    let userRole = await entityManager.findOne(UserRole, {
-      where: {
-        user: this,
-        role,
-      },
-    });
+  async addRole(manager: EntityManager, role: Role, indexPrefix?: string): Promise<User> {
+    let userRole = await this.findUserRole(role);
```

Comment:
The primary changes to this file were to fix the uninitialized `userRoles` relation, and to `push` and `splice` array entries appropriately.